components/telemetry-operator/controller/logpipeline/controller.go

```diff
@@ -54,6 +57,11 @@ type DaemonSetAnnotator interface {
 	SetAnnotation(ctx context.Context, name types.NamespacedName, key, value string) error
 }
 
+type GlobalConfigHandler interface {
```

Comment:
Let's move this interface to the overrides package (same for tracepipeline/controller.go)